astropy/uncertainty/core.py

```diff
@@ -311,6 +311,9 @@ def view(self, dtype=None, type=None):
 
     # Override __getitem__ so that 'samples' is returned as the sample class.
     def __getitem__(self, item):
+        if isinstance(item, Distribution):
```

Comment:
I thing it might be better to use duck-typing here - will make a follow-on PR with a proposal

Reply:
after looking more closely I realize this is a bigger topic that's not worth worrying about right now, because it would be more consistent to do this with *all* the `isinstance` calls in this sub-package, which might have unintended consequences...

Reply:
Indeed. While usually a big fan of duck-typing, I think in this instance, `isinstance` may actually be OK; e.g., I don't think we can count on other implementations to use `.distribution` as an attribute... But something to think about nevertheless...